demos/modals/ModalAuthentication.tsx

```diff
@@ -13,16 +13,24 @@ import Input from '@system/Input';
 import OutsideElementEvent from '@system/detectors/OutsideElementEvent';
 
 import { FormHeading, FormSubHeading, FormParagraph, InputLabel } from '@system/typography/forms';
+import { ModalComponent } from '@root/system/modals/ModalContext';
+import { ViewerContext } from '@components/Page';
 
-export default function ModalAuthentication(props) {
+export interface ModalAuthenticationProps {
+  active?: string;
+}
+
+const ModalAuthentication: ModalComponent<ModalAuthenticationProps> = (props) => {
+  const viewer = React.useContext(ViewerContext);
+  
   const [email, setEmail] = React.useState<string>('');
   const [password, setPassword] = React.useState<string>('');
   const [loading, setLoading] = React.useState<boolean>(false);
 
-  if (props.viewer) {
+  if (viewer) {
     return (
       <div className={styles.wrapper}>
-        <OutsideElementEvent onOutsideEvent={() => props.onShowModal(null)} style={{ width: '100%', maxWidth: 488, margin: `0 auto 0 auto` }}>
+        <OutsideElementEvent onOutsideEvent={() => props.close()} style={{ width: '100%', maxWidth: 488, margin: `0 auto 0 auto` }}>
```

Comment:
`onClose`?

Reply:
i named this without "on" because the user's intended relation to this function is as a caller of the function, not a provider of a callback. in my code i have always prefixed function callbacks that must be defined with "on", while functions that are to be called by the user are not. so i want to make it clear that there is thought behind why i did this. but i will change it.